rfcs/text/0000-query-by-metadata-schema-versions.md

```diff
@@ -0,0 +1,202 @@
+
+In order to enable more frequent metadata schema change-related pushes to production, a [metadata schema integration 
+test design](https://docs.google.com/document/d/18C_Wbrr4Huc8yxQXSdNbzeGk0glBjkfq0MGnozwlIB8/edit) has been proposed. 
+One part of design is to require downstream components to modify their production subscriptions to the Data Store 
+Service (DSS) such that they only receive notifications about bundles with which they are compatible. This will allow
+wranglers to submit new data to the DSS without having to wait for compatibility of the whole system and also allows
+downstream components to receive data independent of whether another component is ready as well. Due to the fact that
+there is currently a metadata schema freeze disallowing any changes, major, minor, or patch, to be pushed to production,
+this document simply designs the minimum system required to unblock wranglers from submitting data and protecting
+downstream components (and therefore the entire DCP) from breaking due to incompatibility.
+
+Note: the subscription modifications will only exist in the production environment. Once the metadata schema integration
+tests are setup, subscriptions will be modified to remove any restrictions on metadata schema versions in order to
+eagerly surface incompatibilities.
+
+
+### User Stories
+
+*Share the [User Stories](https://www.mountaingoatsoftware.com/agile/user-stories) motivating this RFC.*
+* As a data wrangle, I would like to be able to push new data adhering to the latest metadata schema without having to
+wait for every single component to be compatible with the new schema versions.
+* As a DCP developer working on a downstream component (e.g. Matrix Service, Secondary Analysis, etc.) I would like to
+receive notification about new data adhering to new metadata schema versions as soon as I am ready without having to
+wait for other DCP components to also be compatible. 
+
+## Detailed Design
+
+There are two questions fueling the design of this system:
+
+1) How will the schema version be noted in the metadata schema?
+2) Will ElasticSearch, JMESPath, and Query Service all be able to issue range queries against the format of the schema
+version?
+
+Today, the version of the metadata schema is embedded in the `described_by` field which contains the schema version
+embedded in the URL; an example value is “https://schema.humancellatlas.org/type/biomaterial/15.3.0/donor_organism” and
+the schema version is 15.3.0. Unfortunately embedding the version within the URL does not facilitate searching for a
+range of data between schema versions. 
+
+### New schema version fields in metadata schema
+We propose adding two new required fields to each list of schema properties:
+
+```
+"schema_major_version": {
+      "description": "The major version number of the schema.",
+      "type": "integer",
+      "pattern": "^[1-9]\\d*$",
```

Comment:
I think @maniarathi was looking to exclude having "0" be the first digit in an integer for the major version.

Reply:
Correct. Resolving this conversation as there is no action item here.

Reply:
@maniarathi in all fairness, you should leave resolution to the original author of the comment. They might not agree with your assessment that there is no action here.

Reply:
Yes, I missed the intent. Thanks for pointing that out, @malloryfreeberg.

Wouldn't 

```suggestion
      "minimum": 1,
```

be a more straightforward way to achieve this?

Reply:
@hannes-ucsc  1) Apologies for resolving it too quickly. It was simply a habit from my previous job where folks re-opened if necessary. Simply a difference in process :) 2) minimum sounds reasonable to me. I will commit the change.

Reply:
Frankly I agree 100% with closing and re-opening but this seems to upset people. Instead  stuff just gets ignored because few people come back and close their comments. Personally I prefer a transparent close but :man_shrugging: